lib/auth/auth_with_roles.go

```diff
@@ -1449,10 +1449,11 @@ func (a *AuthWithRoles) UpsertRole(role services.Role) error {
 
 // GetRole returns role by name
 func (a *AuthWithRoles) GetRole(name string) (services.Role, error) {
-	if err := a.action(defaults.Namespace, services.KindRole, services.VerbRead); err != nil {
-		// allow user to read roles assigned to them
-		log.Infof("%v %v %v", a.user, a.user.GetRoles(), name)
-		if !utils.SliceContainsStr(a.user.GetRoles(), name) {
+	// Current-user exception: we always allow users to read roles
+	// that they hold.  This requirement is checked first to avoid
+	// misleading denial messages in the logs.
+	if !utils.SliceContainsStr(a.user.GetRoles(), name) {
```

Comment:
This could be potentially dangerous - the role can be complicated or has complex rules in it that were not intended for a user to see, can we remove this part for now?